### PR TITLE
Allow custom cards and more-info

### PR DESCRIPTION
**Pull request by bramkragten**
Allow custom cards for the domains that have "big" cards.

Same config as a "state" card.

### CODE REVIEW

**Comment by andrey-git** [comment]
Paulus,

I think we should drop the requirement in `state-card-content` that custom_ui elements start with "state-card". 
Maybe also rename `custom_ui_state_card` attribute to `custom_ui` and use the value as full element name.

It is very convenient to have a single custom ui element entrypoint, as it can be vulcanized.

**Comment by balloob** [comment]
I agree, we should drop the `state-card` requirement. However, how would you decide if something has to be rendered in an individual card or as part of a group?

**Comment by andrey-git** [comment]
We could override `window.hassUtil.DOMAINS_WITH_CARD` on per-entity basis by an attribute.
Otherwise the card-vs-state-card will remain by domain.

If someone writes a single state card/state-card it's not a problem - they can still create `card-weather.html` or `state-card-custom_light.html`.

If someone creates a single entrypoint - `custom_ui.html` it will need to look into `window.hassUtil.DOMAINS_WITH_CARD` to determine if the "original" element is a card or a state-card.

**Comment by bramkragten** [comment]
The way I made it isn't going to work, because it now also takes the `card` in the `more-info`.
If we would only have a custom_ui attribute, how would you customize both a `state-card` for in the `more-info` and a `card`?
I now made it a separate attribute `custom-card`, so you can define both...

**Comment by andrey-git** [comment]
I don't understand your last comment.
Your custom code can check the `inDialog` property to know if it in `more-info` or not.

If you want separate entry points for similar things - it gets ugly.

Paulus, should we have a separate attributes for card/state-cards?

In any case let's move away from mandatory file prefixes.

**Comment by bramkragten** [comment]
That would make this a breaking-change as of now.

Yes, we could use inDialog, but that would mean a user would have to duplicate the normal state-card in the custom-card if he doesn't want to customize the state-card in the more info.

Or am I missing something?

**Comment by andrey-git** [comment]
I'm fine with the change as it is now, as it still allows to have a single entrypoint.

File: src/util/hass-util.html [hunk 1]
```diff
@@ -405,7 +420,7 @@
   var domain = window.hassUtil.computeDomain(state);
 
   if (unit && domain === 'sensor') {
-    if (unit === '°C' || unit === '°F') {
+    if (unit === 'Â°C' || unit === 'Â°F') {
```
Comment (andrey-git) [inline src/util/hass-util.html]: Extra characters here

File: src/util/hass-util.html [hunk 2]
```diff
@@ -25,7 +25,7 @@
 window.hassUtil.DOMAINS_WITH_MORE_INFO = [
   'alarm_control_panel', 'automation', 'camera', 'climate', 'configurator',
   'cover', 'fan', 'group', 'light', 'lock', 'media_player', 'script',
-  'sun', 'updater', 'vacuum',
```
Comment (andrey-git) [inline src/util/hass-util.html]: You removed 'vacuum' here.

File: src/util/hass-util.html [hunk 3]
```diff
@@ -231,16 +231,31 @@
...
+  if (stateObj.attributes && 'custom_ui_more_info' in stateObj.attributes) {
+    var customUI = stateObj.attributes.custom_ui_more_info;
+    window.hassUtil.ensureCustomUILoaded(customUI);
+    return customUI;
+  }
   var domain = window.hassUtil.computeDomain(stateObj);
 
   if (window.hassUtil.DOMAINS_WITH_MORE_INFO.indexOf(domain) !== -1) {
-    return domain;
+    return 'MORE-INFO-' + domain;
```
Comment (andrey-git) [inline src/util/hass-util.html]: Since the return value of this function need to be uppercased anyway, I think it is more readable to use lower-case here.

**Comment by andrey-git** [comment]
There are conflicts because fetching custom ui html is now done separately.

**Comment by balloob** [comment]
This PR seems to have gone stale. Closing it. You can reopen it when you're ready to finish it.

**Pull request closed without merging**